beacon_node/network/src/sync/manager.rs

```diff
@@ -631,12 +725,13 @@ impl<T: BeaconChainTypes> SyncManager<T> {
             block_roots: vec![parent_hash],
         };
         // select a random fully synced peer to attempt to download the parent block
-        let peer_id = self.full_peers.iter().next().expect("List is not empty");
+        let available_peers = self.full_peers.iter().collect::<Vec<_>>();
+        let peer_id = (**available_peers
+            .choose(&mut rand::thread_rng())
+            .expect("List is not empty"))
```

Comment:
Perhaps you could avoid this `expect` by removing the `self.full_peers.is_empty()` check on L704 and doing this instead:

```rust
let peer_id = if let Some(peer_id) = **available_peers.choose(&mut rand::thread_rng()) {
  peer_id.clone()
} else {
  return;
}
```

Reply:
There's also an `expect` on L722 that I assume would be more helpful if it raised a `CRIT`.